### PR TITLE
FIX: pseudopos readback desync issues

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

Make `PseudoPositioner` devices defined in this module only write to the `ophyd_readback` signal if they were the most recent device to write to the `ophyd_setpoint` signal.

## Motivation and Context

If multiple python sessions are run with slightly different versions of the same pseudo positioner, they can fight over control of the ophyd_readback signal.

see https://jira.slac.stanford.edu/browse/LCLSECSD-1761

## How Has This Been Tested?

I added a unit test here. This should also be tested in a hutch. It's possible that I'll need to make some tweaks.

## Where Has This Been Documented?

Adding some release notes

## Pre-merge checklist
- [ ] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by ZLLentz** [comment]
I'll ask for reviews in the new year as to not distract vacation time!

**Review by klauer — APPROVED** [review]
This is pretty simple and looks good to me 👍 

I had a longer comment thinking through how this would work and possible edge cases, but the short of it is that I don't think there's any real concern and this is about as simple/effective as it could get.

**Comment by ZLLentz** [comment]
I think I fixed the test suite issue that I caused, it might take some time to be allowed test this on real hardware for edge cases but I'm pretty confident that the testing is sufficient.

**Review by tangkong — APPROVED** [review]
This looks good to me too.  I was halfway to reviewing this earlier but got distracted, sorry

File: pcdsdevices/pseudopos.py [hunk 1]
```diff
@@ -221,6 +230,35 @@ def set_current_position(self, position):
...
+        for positioner in self._pseudo:
+            try:
+                signal = getattr(positioner, 'notepad_setpoint', None)
+                if signal is None:
+                    continue
+                signal.subscribe(self._our_move_check)
+            except Exception as ex:
+                self.log.debug(
+                    'Failed to set up _our_move_check subscriptions.',
+                    exc_info=ex,
```
Comment (ZLLentz) [inline pcdsdevices/pseudopos.py:259]: Haha one last small thing on self-review
This is a bool kwarg and I'm passing an exception instance
This "works" but feels wrong

Comment (ZLLentz) [reply pcdsdevices/pseudopos.py:259]: TIL this actually also optionally takes an exception instance which in this case does the same thing but in other cases may not

Comment (klauer) [reply pcdsdevices/pseudopos.py:259]: Right, for reference - https://docs.python.org/3/library/logging.html
```
Changed in version 3.5: The exc_info parameter can now accept exception instances.
```
